hydra/core/utils.py

```diff
@@ -82,17 +87,32 @@ def filter_overrides(overrides: Sequence[str]) -> Sequence[str]:
     return [x for x in overrides if not x.startswith("hydra.")]
 
 
+def _get_callbacks_for_run_job(hydra_context: Optional[HydraContext]) -> "Callbacks":
+    if hydra_context is None:
+        warnings.warn(
+            message="\n"
+            "\trun_job's signature has changed in Hydra 1.1.\n"
+            "\tSupport for the old style will be removed in Hydra 1.2.\n"
+            "\tFor more info, check https://github.com/facebookresearch/hydra/pull/1581.\n"
+        )
+        from hydra._internal.callbacks import Callbacks
+
+        callbacks = Callbacks()
+    else:
+        callbacks = hydra_context.callbacks
+
+    return callbacks
+
+
 def run_job(
-    *,
-    hydra_context: HydraContext,
     task_function: TaskFunction,
     config: DictConfig,
     job_dir_key: str,
     job_subdir_key: Optional[str],
     configure_logging: bool = True,
+    hydra_context: Optional[HydraContext] = None,
 ) -> "JobReturn":
-
-    callbacks = hydra_context.callbacks
+    callbacks = _get_callbacks_for_run_job(hydra_context)
```

Comment:
I don't really consider run_job public API people should use (except plugin authors).
What is the purpose of printing a warning from it?
It looks like plugin authors will see the warning telling them about the change for their own plugin.
and hopefully the generic plugin tests can confirm that they are compliant. 

am I missing anything?

Reply:
Launcher authors will call `run_job` in their launcher impl - so I think it'd make sense to 
issue a warning: when they upgrade their plugins they will make sure to pass in HydraContext in 
run_job. In Hydra 1.2, we will be able to make HydraContext requried in `run_job`. Thoughts?

Reply:
Makes sense.